ansible/roles/prosody/files/mod_rate_limit.lua

```diff
@@ -118,11 +118,17 @@ end
 -- Throttles reading from the connection of a specific session.
 local function throttle_session(session)
 	if not session.jitsi_throttle then
-		module:log("info", "Enabling throttle (%s bytes/s), session=%s, ip=%s.", config.session_rate, session, session.ip);
-		session.jitsi_throttle = new_throttle(config.session_rate, 2);
-		-- throttle.start used for stop throttling after the timeout
-		session.jitsi_throttle.start = gettime();
-		filters.add_filter(session, "bytes/in", limit_bytes_in, 1000);
+        if (session.conn and session.conn.setlimit) then
+            -- TODO: we don't have a mechanism to unthrottle a session in this case.
+		    module:log("info", "Enabling throttle (%s bytes/s) via setlimit, session=%s, ip=%s.", config.session_rate, session, session.ip);
+            session.conn:setlimit(config.session_rate);
```

Comment:
There was a setLimit(0) in the original prosody module for unthrotle.

Reply:
I don't see a place hook into to *detect* that a session should be unthrottled.

Reply:
You are right the setLimit(0) is used for whitelisting only, sorry.